client/state/shortcodes/reducer.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { merge } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { shortcodesSchema } from './schema';
+import { createReducer } from 'state/utils';
+import {
+	SHORTCODE_RECEIVE,
+	SHORTCODE_REQUEST,
+	SHORTCODE_REQUEST_FAILURE,
+	SHORTCODE_REQUEST_SUCCESS
+} from 'state/action-types';
+
+const requestingShortcode = ( state, siteId, shortcode, requesting ) => {
+	return merge( {}, state, {
+		[ siteId ]: {
+			[ shortcode ]: requesting
+		}
+	} );
+};
+
+/**
+ * Returns the updated requests state after an action has been dispatched. The
+ * state maps site ID keys to the object with the site's shortcodes. Each shortcode
+ * is true if it's being currently requested, false otherwise.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const requesting = createReducer( {}, {
```

Comment:
This is fine, but you could avoid some of the boilerplate of passing arguments by having the function return the reducer handler:

```js
const createRequestingReducer = ( requesting ) => {
	return ( state, { siteId, shortcode } ) => {
		return merge( {}, state, {
			[ siteId ]: {
				[ shortcode ]: requesting
			}
		} );
	};
};

export const requesting = createReducer( {}, {
	[ SHORTCODE_REQUEST ]: createRequestingReducer( true ),
	[ SHORTCODE_REQUEST_FAILURE ]: createRequestingReducer( false ),
	[ SHORTCODE_REQUEST_SUCCESS ]: createRequestingReducer( false ),
} );
```